pyproject.toml

```diff
@@ -1,3 +1,11 @@
+[tool.versioneer]
```

Comment:
This PR contains a commit that upgrades versioneer and moves the config to pyproject.toml, and then another commit that gets rid of it. In principle, I would not mind merging it like that, but if it is not too much work to disentangle these changes from the other changes done to pyproject.toml in between, maybe consider squashing both commits into a single commit that removes it directly?